Libplanet.Store/BaseStore.cs

```diff
@@ -46,35 +45,37 @@ public abstract class BaseStore : IStore
             BlockHash branchpoint
         );
 
-        public abstract Transaction GetTransaction(TxId txid);
+        public abstract Transaction? GetTransaction(TxId txid);
 
         public abstract void PutTransaction(Transaction tx);
 
         /// <inheritdoc/>
         public abstract IEnumerable<BlockHash> IterateBlockHashes();
 
         /// <inheritdoc/>
-        public Block GetBlock(BlockHash blockHash)
+        public Block? GetBlock(BlockHash blockHash)
         {
             if (GetBlockDigest(blockHash) is BlockDigest blockDigest)
             {
                 BlockHeader header = blockDigest.GetHeader();
-                (Types.Tx.TxId TxId, Transaction Tx)[] txs = blockDigest.TxIds
-                    .Select(bytes => new Types.Tx.TxId(bytes.ToArray()))
-                    .OrderBy(txid => txid)
-                    .Select(txid => (txid, GetTransaction(txid)))
-                    .ToArray();
-
-                Types.Tx.TxId[] missingTxIds =
-                    txs.Where(pair => pair.Tx is null).Select(pair => pair.TxId).ToArray();
-                if (missingTxIds.Any())
+                TxId[] txids = blockDigest.TxIds
+                                .Select(bytes => new TxId(bytes.ToArray()))
+                                .OrderBy(txid => txid)
+                                .ToArray();
+                Transaction[] txs = txids.Select(txid => GetTransaction(txid))
+                                         .Where(tx => tx != null)
+                                         .OfType<Transaction>()
+                                         .ToArray();
```

Comment:
Probably one last thing. I'm pretty sure `Where()` is unnecessary. 

```suggestion
                Transaction[] txs = txids.Select(txid => GetTransaction(txid))
                                         .OfType<Transaction>()
                                         .ToArray();
```